android/src/test/java/com/revenuecat/purchases/hybridcommon/CommonKtTests.kt

```diff
@@ -33,4 +44,199 @@ internal class CommonKtTests {
             setProxyURLString("this is not a url")
         }
     }
-}
\ No newline at end of file
+
+    private val mockApplicationContext = mockk<Application>(relaxed = true)
+    private val mockContext = mockk<Context>(relaxed = true)
+    private val mockPurchases = mockk<Purchases>()
+
+    @BeforeEach
+    fun setup() {
+        mockkObject(Purchases)
+        every {
+            Purchases.configure(
+                context = any(),
+                apiKey = any(),
+                appUserID = any(),
+                observerMode = any(),
+                service = any()
+            )
+        } returns mockPurchases
+        every { mockContext.applicationContext } returns mockApplicationContext
+    }
+
+    @Test
+    fun `calling logIn correctly passes call to Purchases`() {
+        val appUserID = "appUserID"
+
+        configure(
+            context = mockContext,
+            apiKey = "api_key",
+            appUserID = "appUserID",
+            observerMode = true,
+            platformInfo = PlatformInfo("flavor", "version")
+        )
+
+        every { Purchases.sharedInstance } returns mockPurchases
+        every { mockPurchases.logIn(appUserID, any()) } just runs
+
+        logIn(appUserID = appUserID, onResult = object : OnResult {
+            override fun onReceived(map: Map<String?, *>?) {}
+            override fun onError(errorContainer: ErrorContainer) {}
+        })
+
+        verify(exactly = 1) { mockPurchases.logIn(appUserID, any()) }
+    }
+
+    @Test
+    fun `calling logIn correctly calls onReceived`() {
+        val appUserID = "appUserID"
+
+        configure(
+            context = mockContext,
+            apiKey = "api_key",
+            appUserID = "appUserID",
+            observerMode = true,
+            platformInfo = PlatformInfo("flavor", "version")
+        )
+
+        every { Purchases.sharedInstance } returns mockPurchases
+
+        val mockInfo = mockk<PurchaserInfo>(relaxed = true)
+        val mockCreated = Random.nextBoolean()
+
+        val logInCallback = slot<LogInCallback>()
+        every {
+            mockPurchases.logIn(
+                newAppUserID = appUserID,
+                capture(logInCallback)
+            )
+        } just runs
+
+        val onResult = mockk<OnResult>()
+        every { onResult.onReceived(any()) } just runs
+        every { onResult.onError(any()) } just runs
```

Comment:
You could also do `mockk<OnResult>(relaxed = true)` here if you don't really care about any parameter. Your call. I tend to avoid `relaxed` because it can have unwanted results due to the default values it mocks, but in this case I think it's fine since the behaviour of the mock doesn't really matter for the test and you're just checking is being called in a way.